sds_data_manager/stacks/ialirt_ingest_lambda_stack.py

```diff
@@ -0,0 +1,158 @@
+            ),
+            # Define the read and write capacity units.
+            # TODO: change to provisioned capacity mode in production.
+            billing_mode=ddb.BillingMode.PAY_PER_REQUEST,  # On-Demand capacity mode.
+        )
+        return table
+
+    def create_lambda_function(
+        self, ialirt_bucket: aws_s3.Bucket, packet_data_table: aws_dynamodb.Table
+    ) -> lambda_alpha_.PythonFunction:
+        """Create and return the Lambda function."""
+        lambda_role = iam.Role(
+            self,
+            "IalirtIngestLambdaRole",
+            assumed_by=iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                ),
+                iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "AmazonDynamoDBFullAccess"
+                ),
+            ],
+        )
+
+        lambda_role.add_to_policy(
+            iam.PolicyStatement(
+                actions=[
+                    "s3:GetObject",
+                ],
+                resources=[
+                    packet_data_table.table_arn,
+                    f"{ialirt_bucket.bucket_arn}/*",
+                ],
+            )
+        )
+
+        ialirt_ingest_lambda = lambda_alpha_.PythonFunction(
+            self,
+            id="IalirtIngestLambda",
+            function_name="ialirt-ingest",
+            entry=str(
+                pathlib.Path(__file__).parent.joinpath("..", "lambda_code").resolve()
+            ),
+            index="IAlirtCode/ialirt_ingest.py",
+            handler="lambda_handler",
+            runtime=lambda_.Runtime.PYTHON_3_12,
+            timeout=cdk.Duration.minutes(1),
+            memory_size=1000,
+            role=lambda_role,
+            environment={
+                "TABLE_NAME": packet_data_table.table_name,
+                "S3_BUCKET": ialirt_bucket.bucket_name,
+            },
+        )
+
+        packet_data_table.grant_read_write_data(ialirt_ingest_lambda)
+
+        # The resource is deleted when the stack is deleted.
+        ialirt_ingest_lambda.apply_removal_policy(cdk.RemovalPolicy.DESTROY)
+
+        return ialirt_ingest_lambda
+
+    def create_event_rule(
+        self,
+        ialirt_bucket: aws_s3.Bucket,
+        ialirt_ingest_lambda: lambda_alpha_.PythonFunction,
+    ) -> None:
+        """Create the event rule to trigger Lambda on S3 object creation."""
+        ialirt_data_arrival_rule = events.Rule(
+            self,
+            "IalirtDataArrival",
+            rule_name="ialirt-data-arrival",
+            event_pattern=events.EventPattern(
+                source=["aws.s3"],
+                detail_type=["Object Created"],
+                detail={
+                    "bucket": {"name": [ialirt_bucket.bucket_name]},
+                    "object": {"key": [{"prefix": "packets/"}]},
```

Comment:
Do we want to mimic the main IMAP bucket naming?
```suggestion
                    "object": {"key": [{"prefix": "imap/ialirt/l0/"}]},
```

Reply:
Our s3 bucket is named ialirt-<acct#>. It seems odd to add imap/ialirt as a directory since everything in the bucket will be ialirt. I think packets is fine. It is what Steve Monk said he wanted and I don't think it will confuse anyone.